### ISSUE TITLE
Feature request: Support using URL instance as matcher

### ISSUE THREAD

**Issue by redaxmedia** [wheresrhys/fetch-mock#445]
Using `URL` is causing `indexOf` error on `node-fetch` 7.3.9: 

```
TypeError: matcher.indexOf is not a function
```

I had to workaround using `toString()`:

```
const urlLogin: URL = new URL(environment.routes.authentication.login, environment.apiUrl);
const urlLogout: URL = new URL(environment.routes.authentication.logout, environment.apiUrl);

fetchMock
    .post(urlLogin.toString(), {
        body: {},
        status: 200
    })
    .delete(urlLogout.toString(), {
        body: {},
        status: 200
    });
```

**Comment by wheresrhys** [comment]
Which environment is this in: browser or node. And what's the build setup - are there any polyfills present?

**Comment by redaxmedia** [comment]
This is an spec file running via Jest. There are no fetch polyfills used.

**Comment by wheresrhys** [comment]
I just realised you're using fetch-mock in an undocumented way. The api has never claimed to support URL's as matchers. It's maybe not a bad idea to do so, though as you can get around it by using toString(), I don't see it as a big deal that it's not supported.

Can you point to anywhere where the documentation is unclear about this as I will try to clarify.

**Comment by redaxmedia** [comment]
Didn't you add support for it one year ago? I found  https://github.com/wheresrhys/fetch-mock/issues/355 and therefore used URL.

**Comment by wheresrhys** [comment]
Ah, subtle difference. I was fixing the bug where by e.g. `fetch(URL('https://here.com'))` would not be matched by `fetchMock.mock('https://here.com', 200)`. I think at the time I only picked up on the bug report and not that there was a feature request too.

**Comment by redaxmedia** [comment]
I see... take a night to sleep about supporting URL in general. It is so much cleaner than strings. I am okay with closing this issue also... thanks for the library and your work.

**Comment by wheresrhys** [comment]
This is released in v8.1.0
